lib/kafka_producer.js

```diff
@@ -262,16 +296,35 @@ KafkaProducer.prototype.getWholeMsg = function getWholeMsg(topic, message, timeS
 
 KafkaProducer.prototype.close = function close(callback) {
     var self = this;
+
+    self.enable = false;
+    self.closing = true;
+    self.closeCallback = callback;
+
+    if (self.flushInterval) {
+        clearInterval(self.flushInterval);
+    }
+
     if (self.batching) {
-        self.flushEntireCache(callback);
+        self.flushEntireCache();
```

Comment:
good point. since it's implemented we can leave the feature there for anyone else that wants to flush & wait.